Source/src/components/ComponentTransform.cpp

```diff
@@ -155,9 +161,144 @@ void Hachiko::ComponentTransform::SetLocalRotationInEulerAngles(const float3& ne
     rotation_local = rotation_amount * (rotation_local);
     rotation_euler_local = new_rotation_euler_local;
 
-    UpdateTransformOfHierarchy(MatrixCalculationMode::GLOBAL_FROM_LOCAL);
+    dirty = true;
+}
+
+/***    GETTERS     ***/
+
+const float3& Hachiko::ComponentTransform::GetPosition()
+{
+    if (dirty)
```

Comment:
I would put the` if (dirty)` inside the `UpdateTransformAndChildren` function call so it will always use the dirty flag when trying to update from anywhere. You'll always need to remember to add the check on getters otherwise

Reply:
In theory when a parent becomes dirty all the children must be set as dirty

Reply:
You are right, thanks for bringing this to our attention. Let's review this pr again when the needed changes made to this issue, because of that I dismissed my review that approved the current functionality.

Reply:
The problem I see with this is that `UpdateTransformAndChildren` is a function that is also called when we edit the **global transform** too, but in this case we want to skip the dirty flag and update the local transform of all objects associated to that global at that moment. That's why it's checked outside but only when reading Global data, as Local data can never be dirty.

Reply:
That is correct, local data getters remain inlined and unchanged. All the getters I changed are global ones.